crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1137,17 +1137,20 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
         }
 
         // Make an iterator to update the locks of the transaction's shared objects.
-        let ids = certificate.shared_input_objects();
+        let ids = certificate.shared_input_objects().map(|(id, _)| id);
         let versions = self.tables.next_object_versions.multi_get(ids)?;
 
         let ids = certificate.shared_input_objects();
         let (sequenced_to_write, schedule_to_write): (Vec<_>, Vec<_>) = ids
             .zip(versions.iter())
-            .map(|(id, v)| {
-                // If it is the first time the shared object has been sequenced, assign it the default
-                // sequence number (`OBJECT_START_VERSION`). Otherwise use the `scheduled` map to
-                // to assign the next sequence number.
-                let version = v.unwrap_or_else(|| OBJECT_START_VERSION);
+            .map(|((id, shared_at_version), v)| {
```

Comment:
confused how this compiles - since you earlier added the `map(|(id, _)| id)` call, it seems like you've already removed the `shared_at_version` from `ids`. Maybe you can change the variable names in the map call to make it clearer?

Reply:
yeah maybe changing the name of the second `let ids` would be helpful

Reply:
Agreed, will look into making this clearer.